share/p2p/discovery/discovery.go

```diff
@@ -202,21 +204,32 @@ func (d *Discovery) Advertise(ctx context.Context) {
 // It initiates peer discovery upon request and restarts the process until the soft limit is
 // reached.
 func (d *Discovery) discoveryLoop(ctx context.Context) {
-	t := time.NewTicker(discoveryRetryTimeout)
-	defer t.Stop()
+	backoffDuration := backoffInitialDuration
+	backoffTicker := time.NewTicker(backoffDuration)
+	defer backoffTicker.Stop()
```

Comment:
Not an issue here, but after making changes, pls make sure `Stop()` will not be called twice.